compiler/rustc_expand/src/mbe/macro_rules.rs

```diff
@@ -485,7 +485,9 @@ pub fn compile_declarative_macro(
                     )
                     .pop()
                     .unwrap();
-                    valid &= check_lhs_nt_follows(sess, def, &tt);
+                    // We don't handle errors here, the driver will abort
+                    // after parsing/expansion. we can report every error in every macro this way.
+                    valid &= check_lhs_nt_follows(sess, def, &tt).is_ok();
```

Comment:
Pre-existing, but updating `valid` here is pointless when it's immediately followed by `return`.

Reply:
Which means `check_lhs_nt_follows` doesn't need to return a `bool`.

It would be good to know how/why the driver will abort after parsing/expansion, bonus points if you understand that and can augment this comment appropriately.

Reply:
No, this return is in a closure

Reply:
👍